src/lib.rs

```diff
@@ -155,17 +131,17 @@ impl Iso8583File {
     /// containing only the searched fields.
     /// This process is memory intensive, due to the imutable nature of this method
     pub fn search(self, search: HashMap<String, Vec<String>>) -> Iso8583File {
-        let mut search_groups_result: Vec<Group> = vec![];
+        let mut search_messages_result: Vec<Message> = vec![];
 
         //TODO fix the performance for the search
-        // since this loops inside each group and uses a hash to find a match, this is memory intensive
-        for group in self.groups {
-            let group_messages_hash = group.get_messages_hash().expect("Unable to find message hash for group. Maybe the file is broken");
+        // since this loops inside each message and uses a hash to find a match, this is memory intensive
+        for message in self.messages {
+            let message_hash = message.get_messages_hash().expect("Unable to find message hash for message. Maybe the file is broken");
             for search_key in search.keys() {
-                match group_messages_hash.get(search_key) {
-                    Some(message) => {
-                        if search.get(search_key).unwrap().contains(message) {
-                            search_groups_result.push(group);
+                match message_hash.get(search_key) {
```

Comment:
Since, we'll remove some legacy code we'll have to adapt this to consume the new structs.